kong/utils.go

```diff
@@ -741,3 +741,147 @@ func FillPluginsDefaults(plugin *Plugin, schema Schema) error {
 func FillPluginsDefaultsWithOpts(plugin *Plugin, schema map[string]interface{}, opts FillRecordOptions) error {
 	return fillConfigRecordDefaultsAutoFields(plugin, schema, opts)
 }
+
+// deleteAndCollapseMap is a utility function that removes element from a map using path.
+// If it turns out that that was the last key in the map then it collapses map key itself.
+// Example:
+//
+//		configMap := map[string]interface{}{
+//			"a": map[string]interface{}{
+//				"a1": 1,
+//			},
+//	   "b": 456,
+//		}
+//
+// then by calling deleteAndCollapseMap(configMap, []string{"a", "a1"})
+// // the result would be: (entire "a" key is deleted as well)
+//
+//	map[string]interface{} {
+//	  "b": 456,
+//	}
+func deleteAndCollapseMap(config map[string]interface{}, path []string) {
+	if len(path) == 0 {
+		return
+	}
+
+	key := path[0]
+	if len(path) == 1 {
+		delete(config, key)
+		return
+	}
+
+	if nested, ok := config[key].(map[string]interface{}); ok {
+		deleteAndCollapseMap(nested, path[1:])
+		if len(nested) == 0 {
+			delete(config, key)
+		}
+	}
+}
+
+// same as ClearUnmatchingDeprecations but this function below is adjusted for recursive use (which is required
+// when the schema contains nested records and they can have nested shorthands).
+func clearUnmatchingDeprecationsHelper(
+	newPluginConfig Configuration,
+	oldPluginConfig Configuration,
+	schema *gjson.Result,
+) {
+	configFields := schema.Get("fields")
+	// Fetch deprecated fields
+	shortHandFields := schema.Get("shorthand_fields")
+
+	shortHandFields.ForEach(func(_, value gjson.Result) bool {
+		field := value.Map()
+		for deprecatedFieldName, shorthandFieldConfig := range field {
+			if deprecatedFieldValue, ok := newPluginConfig[deprecatedFieldName]; ok {
+				// deprecatedFieldName is used in new plugin configuration
+				// verify if the fields that this depractedField is replaced with
+				// are also sent in new plugin configuration - if not clear them from old plugin configuration
```

Comment:
this comment does not seem to refer to the code right below it, and the same logic is explained again below, I would remove these 2 lines of comment.

Reply:
Fixed ✅